src/libs/opts/opts.c

```diff
@@ -0,0 +1,951 @@
+#endif
+
+static inline const char * keyGetMetaString (const Key * key, const char * meta)
+{
+	const Key * mk = keyGetMeta (key, meta);
+	const char * value = mk == NULL ? NULL : keyString (mk);
+	return value != NULL && value[0] == '\0' ? NULL : value;
+}
+
+static int addProcKey (KeySet * ks, const Key * key, Key * valueKey);
+static KeySet * parseEnvp (const char ** envp);
+
+static KeySet * parseArgs (KeySet * optionsSpec, int argc, const char ** argv, Key * errorKey);
+static void setOption (Key * option, const char * value, bool repeated);
+
+static Key * splitEnvValue (const Key * envKey);
+
+static KeySet * ksMetaGetSingleOrArray (Key * key, const char * metaName);
+
+static char * generateHelpMessage (KeySet * optionsSpec, const char * progname);
+
+/**
+ * This functions parses a specification of program options, together with a list of arguments
+ * and environment variables to extract the option values.
+ *
+ * The options have to be defined in the metadata of keys in the spec namespace. If an option value
+ * is found for any of the given keys, a new key with the same path but inside the proc namespace
+ * will be inserted into @p ks. This enables a cascading lookup to find these values.
+ *
+ * If @p argv contains "-h" or "--help" @p ks will not be changed, instead the value of @p errorKey
+ * will be set to a help message describing available options and 1 will be returned. The program
+ * name used in this message is taken from `argv[0]`. If it contains a '/' only the part after the
+ * last '/' will be used. The help message will contain description for each option. This description
+ * is taken from `opt/help`, or if `opt/help` is not specified from `description`. If neither `opt/help`
+ * nor `description` are given, or `opt/nohelp` is set to "1", the option will not appear in the help
+ * message. The same applies to `env`s.
+ *
+ * The help message can also be customised in two more ways:
+ * 1. By setting the `help/usage` metadata on @p errorKey, you can use a custom usage line for the help
+ *    message.
+ * 2. By setting the `help/prefix` metadata on @p errorKey, you can insert text between the usage line
+ *    and the description of options.
+ * There is no `help/suffix` metadata, because you can easily append text to the help message yourself.
+ *
+ * Because "-h" and "--help" are reserved for the help message, neither can be used for anything else.
+ *
+ * To define a command line option set the `opt` meta-key to the short option. Only the first
+ * character of the given value will be used ('\0' is not allowed). You can also set `opt/long`,
+ * to use a long option. Each option can have a short version, a long version or both.
+ * Per default an option is expected to have an argument. To change this behaviour set `opt/arg` to
+ * either 'none' or 'optional' (the default is 'required'). The behaviour of short options, long
+ * options, required and optional arguments is similar to getopt_long(3).
```

Comment:
similar? What are the differences? (as described below?)

Reply:
If all differences are listed, then simply say "as described below". If users should not assume that the behavior is as in `getopt_long`, you better not mention it in such specific statements.